### DISCUSSION TITLE
Is it possible to have a corner radius on a Label's bg_color?

### DISCUSSION THREAD

**Discussion by kelltom** [TomSchimansky/CustomTkinter#214] [Q&A] [answered]
I'd like to round the corners of this label, but the _corner_radius_ argument is not working in this case. Here's the label with a long string of text and a light-grey background color.

![image](https://user-images.githubusercontent.com/44652363/175823473-972392fb-a450-483b-b572-f5d3a7731a12.png)

Here's the code for the label you see above.

![image](https://user-images.githubusercontent.com/44652363/175823691-fc899a99-01f1-4101-97ac-695b90024e04.png)

Here's what it looks like with a corner_radius of 40. Note how it only applies this radius directly to the invisible bounding box of the text itself, not the entire label filling the grid cell. This essentially just severs some of the text.

![image](https://user-images.githubusercontent.com/44652363/175823750-44e9d991-775d-4eed-91e7-843eaa121081.png)

Is there any way to apply a border-radius to the bg_color?

**Comment by Valenterie** [comment, marked as answer]
Hey ! Had the same issue the other day, what you can try to do is just not to use a label, use the "canvas.create_..." function, it can work with arc or oval, maybe rectangle but i'm not sure. You have to make the text separately though.

**Reply by kelltom** [reply]
Sounds like that should do it, thanks!

**Reply by Valenterie** [reply]
What you can do is create a function were you create two rectangles and then 4 arcs or ovals : 

![exemple](https://user-images.githubusercontent.com/87931771/175982623-ddb56311-2d51-43c0-859d-7d8609684f3a.png)

Put that in a function like "create_rectangleBorder" with 2 times "name.create_rectagle" and then 4 times "name.create_arc" or "name.create_oval", don't know which one works the best

Good luck !

**Reply by Valenterie** [reply]
If you need it, i'm currently trying to do that function myself for my project.

**Reply by TomSchimansky** [reply]
You know this is already implemented for the CTkLabel?

**Reply by TomSchimansky** [reply]
The CTkLabel already has a corner_radius attribute, you can simply use it. I mean you can see it in this screenshot: https://github.com/TomSchimansky/CustomTkinter/blob/master/documentation_images/Windows_dark.png

**Reply by TomSchimansky** [reply]
As I already wrote you have to use fg_color for the main label color instead of bg_color. bg_color is something different than bg in tkinter, look at this: https://github.com/TomSchimansky/CustomTkinter/wiki/Themes#colors

**Reply by Valenterie** [reply]
Ohh okay so no need to do all that stuff lmao

**Reply by kelltom** [reply]
> As I already wrote you have to use fg_color for the main label color instead of bg_color. bg_color is something different than bg in tkinter, look at this: https://github.com/TomSchimansky/CustomTkinter/wiki/Themes#colors

Gotcha, but the problem I outlined arises when dealing with wrapping text upon resizing the view. 
![python_etVb41Boz1](https://user-images.githubusercontent.com/44652363/176211894-110b0c8c-ef67-435d-807d-c7e3ebeef2c7.gif)

This is with a fg_color defined, no bg_color, and no static height defined.

**Reply by Valenterie** [reply]
I don't know how you made the text adapt to the size of the window, but maybe if you find a way to do it also to the label and then putting the text on it, it would work.

**Reply by kelltom** [reply]
> I don't know how you made the text adapt to the size of the window, but maybe if you find a way to do it also to the label and then putting the text on it, it would work.

That is done with this line here:
![image](https://user-images.githubusercontent.com/44652363/176220338-a749b39f-576b-4c1d-bd75-902e96878bb4.png)

`self.lbl_script_desc.bind('<Configure>', lambda e: self.lbl_script_desc.config(wraplength=self.lbl_script_desc.winfo_width()-10))`

It's basically resetting the wraplength property of the label either every time it changes, or every iteration of the main loop (not too sure which, but it works either way). I tried to also configure the height/width of the label this way (by setting it to the height of the grid cell), but it just prevented the wrapping from working - so I suppose I don't fully understand how it's working. Logically though, it seems that it should be the solution - to constantly be updating the height and width of the label such that it fills the grid cell. 

For the time being, I've just removed the fg_color entirely so it's just white text on the dark grey frame and it looks fine enough, so I'm not beating myself up over it for now, but it would be good to know for the future.

**Comment by TomSchimansky** [comment]
You have to use fg_color for the main color of the label, bg_color is the color behind the label and you don't have to set it manually:
https://github.com/TomSchimansky/CustomTkinter/wiki/Themes#colors
